code/IntercomCleanupTask.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Build task to delete users in Intercom who do not exist in the local database.
+ */
+class IntercomCleanupTask extends \BuildTask {
+
+	/**
+	 * @var GuzzleHttp\Client
+	 */
+	protected $client;
+
+	/**
+	 * @var \DataList
+	 */
+	protected $members;
+
+	/**
+	 * @var int
+	 */
+	protected $deleted = 0;
+
+	/**
+	 * @var bool
+	 */
+	protected $dryRun = false;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->client = \Injector::inst()->get('Sminnee\SilverStripeIntercom\Intercom')->getClient();
+		$this->members = \Member::get();
+	}
+
+	/**
+	 * @param \SS_HTTPRequest $request
+	 */
+	public function run($request) {
+		if ($request && $request->requestVar('dryrun')) {
+			$this->dryRun = true;
+		}
+
+		if ($this->dryRun) {
+			$this->log('Running in dry-run mode. No users will be deleted');
+		}
+
+		// the first page has users included in the response, so we'll process those to start with
+		$response = $this->client->users->getUsers(['page' => '1']);
+		foreach ($response->users as $user) {
+			$this->processUser($user);
+		}
+
+		// process users of each page response one at a time
```

Comment:
is there a risk that we are missing users since you delete them above and unprocessed users would "fall down" into page 1 and therefore not processed?